### PR TITLE
feat: add cart function to bigcommerce example

### PR DESCRIPTION
**Pull request by joshuawootonn**
This includes adding cart function and updating the readme accordingly.

### CODE REVIEW

File: examples/bigcommerce/pages/api/categories.ts [hunk 1]
```diff
@@ -6,7 +6,6 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     const categories = await getCategories()
     res.status(200).json(categories)
   } catch (error) {
-    console.error('Failed to fetch categories', error)
-    res.status(400).json({ message: 'Failed to fetch categories' })
+    res.status(400).json({ message: 'Failed to fetch categories.' })
```
Comment (joshuawootonn) [inline examples/bigcommerce/pages/api/categories.ts:9]: Changed this to match the catch case of the `cart` and `checkout` routes

File: examples/bigcommerce/pages/api/categories.ts [hunk 2]
```diff
@@ -2,11 +2,6 @@ import { getCategories } from 'lib/bigcommerce'
...
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  try {
-    const categories = await getCategories()
-    res.status(200).json(categories)
-  } catch (error) {
-    console.error('Failed to fetch categories', error)
-    res.status(400).json({ message: 'Failed to fetch categories' })
-  }
+  const categories = await getCategories()
```
Comment (joshuawootonn) [inline examples/bigcommerce/pages/api/categories.ts:5]: This is already logging errors, and if it throws this should be a 500, so I moved this to be not include the try catch

File: examples/bigcommerce/components/product-cart.tsx [hunk 3]
```diff
@@ -1,68 +1,243 @@
+import { Popover, Transition } from '@headlessui/react'
+import Image from 'next/image'
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+
 import { useCart } from 'lib/cart-context'
 import { useProduct } from 'lib/product-context'
+import { Cart34, Minus28, Minus36, Plus28, Plus36, Spinner22, Spinner28 } from './icons'
+
+function formatPrice(value?: number) {
```
Comment (migueloller) [inline examples/bigcommerce/components/product-cart.tsx:10]: Have you heard about the [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) API? I've never used it but it looks pretty cool!

File: examples/bigcommerce/pages/api/cart.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    if (!response.ok) throw new Error(response.statusText)
+
+    if (response.status === 204) return res.status(204).json({})
+
+    const result: RestResponse<CartResponse | null> = await response.json()
+
+    return res.status(200).json(result)
+  }
+
+  return res.status(405)
```
Comment (migueloller) [inline examples/bigcommerce/pages/api/cart.ts:145]: I think that if you don't send the response the request will just hang until the client times out. You might want to add `.send('Method Not Allowed')` or something to make sure the request doesn't hang

**Review by migueloller — DISMISSED** [review]
Looking great!

File: examples/bigcommerce/pages/api/cart.ts (hunk 4 above)
Comment (joshuawootonn) [reply examples/bigcommerce/pages/api/cart.ts:145]: You are correct. Thanks for catching that.

File: examples/bigcommerce/components/product-cart.tsx (hunk 3 above)
Comment (joshuawootonn) [reply examples/bigcommerce/components/product-cart.tsx:10]: Going to pass on this add for now

**Review by migueloller — APPROVED** [review]

**Pull request merged**